Source/EventAnalysis/KTOverlappingTrackClustering.cc

```diff
@@ -0,0 +1,406 @@
+        }
+        return;
+    }
+
+    bool KTOverlappingTrackClustering::Run()
+    {
+        return DoClustering();
+    }
+
+    bool KTOverlappingTrackClustering::DoClustering()
+    {
+        if (! FindMatchingTracks())
+        {
+            KTERROR(otclog, "An error occurred while identifying overlapping tracks");
+            return false;
+        }
+
+        KTDEBUG(otclog, "Track building complete");
+        fDoneSignal();
+
+        return true;
+    }
+
+    bool KTOverlappingTrackClustering::FindMatchingTracks()
+    {
+        KTINFO(otclog, "Finding overlapping tracks");
+        KTDEBUG(otclog, "FrequencyAcceptance is: "<<fMaxTrackWidth);
+        fNewTracks.clear();
+
+        unsigned NumberOfTracks = fCompTracks.size();
+        unsigned NumberOfNewTracks = fNewTracks.size();
+
+
+
+        if (NumberOfTracks > 1)
+        {
+            while (NumberOfTracks!=NumberOfNewTracks)
+            {
+                NumberOfTracks = fCompTracks.size();
+                KTDEBUG(otclog, "Number of tracks to cluster: "<< NumberOfTracks);
+                this->OverlapClustering();
+
+                // Update number of tracks
+                NumberOfNewTracks = fNewTracks.size();
+
+                KTDEBUG(otclog, "Number of new tracks: "<< NumberOfNewTracks);
+
+                fCompTracks.clear();
+                fCompTracks = fNewTracks;
+                fNewTracks.clear();
+            }
+        }
+
+        this->EmitTrackCandidates();
+
+        return true;
+    }
+
+    bool KTOverlappingTrackClustering::OverlapClustering()
+    {
+        bool match;
+        for (std::vector<KTProcessedTrackData>::iterator compIt = fCompTracks.begin(); compIt != fCompTracks.end(); ++compIt)
+        {
+            match = false;
+            for (std::vector<KTProcessedTrackData>::iterator newIt = fNewTracks.begin(); newIt != fNewTracks.end(); ++newIt)
+            {
+                if (this->DoTheyOverlap(*compIt, *newIt))
+                {
+                    match = true;
+                    KTDEBUG(otclog, "Found overlapping tracks")
+                    this->CombineTracks(*compIt, *newIt);
+                    break;
+                }
+
+                // since there appear to be real tracks crossing each other this check is removed here
+                /*if (this->DoTheyCross(*compIt, *newIt))
+                {
+                    match = true;
+                    KTDEBUG(otclog, "Found crossing tracks")
+                    this->CombineTracks(*compIt, *newIt);
+                    break;
+                }*/
+            }
+
+            if (match == false)
+            {
+                KTProcessedTrackData newTrack(*compIt);
+                fNewTracks.push_back(newTrack);
+            }
+        }
+        return true;
+    }
+
+
+    void KTOverlappingTrackClustering::CombineTracks(KTProcessedTrackData& oldTrack, KTProcessedTrackData& newTrack)
```

Comment:
I think this function can be const, and I think `oldTrack` can be a const reference, as well.